resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientUtils.scala

```diff
@@ -94,25 +120,41 @@ private[spark] object KubernetesClientUtils extends Logging {
     val confDir = Option(conf.getenv(ENV_SPARK_CONF_DIR)).orElse(
       conf.getOption("spark.home").map(dir => s"$dir/conf"))
     if (confDir.isDefined) {
-      val confFiles = listConfFiles(confDir.get)
-      logInfo(s"Spark configuration files loaded from $confDir : ${confFiles.mkString(",")}")
-      confFiles.map { file =>
-        val source = Source.fromFile(file)(Codec.UTF8)
-        val mapping = (file.getName -> source.mkString)
-        source.close()
-        mapping
-      }.toMap
+      val confFiles = listConfFiles(confDir.get, conf.get(Config.CONFIG_MAP_MAXSIZE))
+      val filesSeq: Seq[Option[(String, String)]] = confFiles.map { file =>
+        try {
+          val source = Source.fromFile(file)(Codec.UTF8)
+          val mapping = Some(file.getName -> source.mkString)
+          source.close()
+          mapping
+        } catch {
+          case e: MalformedInputException =>
+            logWarning(s"skipped a non UTF-8 encoded file ${file.getAbsolutePath}.", e)
+            None
+        }
+      }
+      val truncatedMap = truncateToSize(filesSeq.flatten, conf.get(Config.CONFIG_MAP_MAXSIZE))
+      logInfo(s"Spark configuration files loaded from $confDir :" +
```

Comment:
Shall we skip this if `truncatedMap` is empty?

Reply:
Or do you want to truncate earlier before reading them all into memory?

Reply:
Challenge with trying to truncate before reading is, we do not know which files will have `MalformedInputException` i.e. they are not encoded with UTF-8.